src/contract_wrappers/modules/transfer_manager/general_transfer_manager_wrapper.ts

```diff
@@ -457,6 +568,220 @@ export default class GeneralTransferManagerWrapper extends ModuleWrapper {
+   * c) Buyer's purchase lockup is over
+   */
+  public executeTransfer = async (params: ExecuteTransfer) => {
+    return (await this.contract).executeTransfer.sendTransactionAsync(
+      params.from,
+      params.to,
+      new BigNumber(100),
+      params.data,
+      params.txData,
+      params.safetyFactor,
+    );
+  };
+
+  /**
+   * Default implementation of verifyTransfer used by SecurityToken
+   */
+  public verifyTransfer = async (params: ExecuteTransfer) => {
+    const result = await (await this.contract).verifyTransfer.callAsync(
+      params.from,
+      params.to,
+      new BigNumber(100),
+      params.data,
+    );
+    return result;
+  };
+
+  /**
+   * Modifies the successful checks required for a transfer to be deemed valid.
+   */
+  public modifyTransferRequirements = async (params: ModifyTransferRequirements) => {
+    const result = await (await this.contract).modifyTransferRequirements.sendTransactionAsync(
+      params.transferType,
+      params.fromValidKYC,
+      params.toValidKYC,
+      params.fromRestricted,
+      params.toRestricted,
+      params.txData,
+      params.safetyFactor,
+    );
+    return result;
+  };
+
+  /**
+   * Modifies the successful checks required for transfers.
+   */
+  public modifyTransferRequirementsMulti = async (params: ModifyTransferRequirementsMulti) => {
+    assert.assert(await this.isCallerAllowed(params.txData, Perm.Admin), 'Caller is not allowed');
+    assert.assert(
+      params.transferTypes.length === params.fromValidKYC.length &&
+        params.fromValidKYC.length === params.toValidKYC.length &&
+        params.toValidKYC.length === params.fromRestricted.length &&
+        params.fromRestricted.length === params.toRestricted.length,
+      'Mismatched input lengths',
+    );
+    return (await this.contract).modifyTransferRequirementsMulti.sendTransactionAsync(
+      params.transferTypes,
+      params.fromValidKYC,
+      params.toValidKYC,
+      params.fromRestricted,
+      params.toRestricted,
+      params.txData,
+      params.safetyFactor,
+    );
+  };
+
+  /**
+   * Add or remove KYC info of an investor.
+   */
+  public modifyKYCDataMulti = async (params: ModifyKYCDataMulti) => {
+    assert.assert(await this.isCallerAllowed(params.txData, Perm.Admin), 'Caller is not allowed');
+    assert.assert(
+      params.investors.length === params.canSendAfter.length &&
+        params.canSendAfter.length === params.canReceiveAfter.length &&
+        params.canReceiveAfter.length === params.expiryTime.length,
+      'Mismatched input lengths',
+    );
+    const canSendAfter: BigNumber[] = [];
+    const canReceiveAfter: BigNumber[] = [];
+    const expiryTime: BigNumber[] = [];
+
+    for (let i = 0; i < params.canSendAfter.length; i += 1) {
+      canSendAfter.push(dateToBigNumber(params.canSendAfter[i]));
+      canReceiveAfter.push(dateToBigNumber(params.canReceiveAfter[i]));
+      expiryTime.push(dateToBigNumber(params.expiryTime[i]));
+    }
```

Comment:
All of the previous section can be rewritten as

```
const candSendAfter = params.canSendAfter.map(dateToBigNumber);
const canReceiveAfter = params.canReceiveAfter.map(dateToBigNumber);
const expiryTime = params.expiryTime.map(dateToBigNumber);
```

And you save yourself the need of a foor loop

Reply:
good catch